Add powershell launch scripts.

d8ahazard opened the pull request:
Add modern equivalents of the webui and webui-user scripts that are more on par with the linux versions.

d8ahazard committed:
Add powershell launch scripts.

AUTOMATIC1111 commented:
having to maintain two sets of scripts is not an improvement over having to maintain one set of scripts

d8ahazard commented:
> having to maintain two sets of scripts is not an improvement over having to maintain one set of scripts

So then delete/deprecate the bat files? I mean, it's only a scripting method that's older than I am. 

Powershell is supported in all versions of Windows since Windows 7, and these scripts bring windows users up to the same standard as those using the linux scripts.

d8ahazard committed:
Merge remote-tracking branch 'upstream/master' into PowerShell

AUTOMATIC1111 commented:
I don't have preference for powershell compared to bat

d8ahazard committed:
Fix env variables not being passed.

For some reason, commandline_args would work, but not torch command. this fixes it so everything sticks.

d8ahazard committed:
Merge remote-tracking branch 'upstream/master' into PowerShell

d8ahazard commented:
So, this is the plan now? Just ignore all my PR's and then close them?

Was there something actually wrong with this?

ClashSAN commented:
@d8ahazard 
added it here, have a look - https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Install-and-Run-on-NVidia-GPUs#powershell

Different topic, but do you happen to know of several distinct features in diffusers that would be of great value in webui? The dreambooth extension as it is currently, would it be difficult or super unoptimized to convert into a diffusers extension? 

I would like to make a comparison/list of feature diffusers will bring to webui. To see if there is any interest. Need to get patrickvonplaten the code additions needed if planning to make an extension. Hopefully this can be done, as during this time auto is more active. 

https://github.com/AUTOMATIC1111/stable-diffusion-webui/discussions/2577#discussioncomment-4448127

d8ahazard commented:
> @d8ahazard added it here, have a look - https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Install-and-Run-on-NVidia-GPUs#powershell
> 
> Different topic, but do you happen to know of several distinct features in diffusers that would be of great value in webui? The dreambooth extension as it is currently, would it be difficult or super unoptimized to convert into a diffusers extension?
> 
> I would like to make a comparison/list of feature diffusers will bring to webui. To see if there is any interest. Need to get patrickvonplaten the code additions needed if planning to make an extension. Hopefully this can be done, as during this time auto is more active.
> 
> [#2577 (reply in thread)](https://github.com/AUTOMATIC1111/stable-diffusion-webui/discussions/2577#discussioncomment-4448127)

Off the top of my head, the biggest one I can think of is LORA. While not a native huggingface feature, LORA lets users save .pt files similar to TI embeddings, except for the pt files contain weights for both the unet and text encoder. 

If there were some way to schlep diffusers back into the UI via extension and load diffusion weights instead of from a .ckpt, then people would be able to merge lora weights on top of the base diffusion model. I know this has been requested a LOT in discussions surrounding my Dreambooth extension. 

And, having looked at the current code in Auto1111 quite a bit for loading ckpt models, I don't think it would be *that* difficult to maybe inject some kind of hook before normal "load_model" is called that could somehow allow using "other" models without an entire rewrite.

Might not work "for all the things" right away, so the hook might need some kind of flag system to indicate what types of inference these new weights could be used for...

EfourC commented:
I am interested in using these powershell scripts to run the WebUI as part of my quest to avoid the slightly annoying "Terminate batch job (Y/N)?" message when killing the current .bat file with Ctrl-c.

However when I tried running `webui-user.ps1`, I ran into an error where the command is rejected due to not being digitally signed.

Some trials showed that if I start a PowerShell 7 console (either directly or within Windows Terminal), and then manually run `.\webui-user.ps1` , I get the error:

> .\webui-user.ps1: File N:\stable-diffusion-webui\webui-user.ps1 cannot be loaded. The file N:\stable-diffusion-webui\webui-user.ps1 is not digitally signed. You cannot run this script on the current system. For more information about running scripts and setting execution policy, see about_Execution_Policies at https://go.microsoft.com/fwlink/?LinkID=135170.

BUT, if I right-click on the file in Explorer and select `Run with PowerShell 7`, it works properly! Unfortunately, I cannot perform any other command line activity this way in the same session, and when I kill the process with Ctrl-C the PowerShell window exits completely immediately, so this isn't how I want to run it.

ExecutionPolicy for CurrentUser and LocalMachine are both set as RemoteSigned.
I don't want to use Unrestricted policy, and doing that creates it's own confirmation prompt each time when running the script anyway.

Any ideas what is going wrong?
**_More importantly for this PR:_**  Would this issue affect other people?

![pwsh_4lQXZc9g11](https://user-images.githubusercontent.com/121468033/224520136-48ab7c02-fa2a-4283-95d3-7a8cbcfa3638.png)
